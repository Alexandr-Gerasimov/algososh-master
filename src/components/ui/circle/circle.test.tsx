import renderer from "react-test-renderer";
import { Circle } from "./circle";
import { render, screen, fireEvent } from "@testing-library/react";
import { ElementStates} from "../../../types/element-states";

describe("Buttons", () => {
  it("Без букв", () => {
    const tree = renderer.create(<Circle />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("С буквами", () => {
    const tree = renderer.create(<Circle letter={""}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("С head", () => {
    const tree = renderer.create(<Circle head={"Head"}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("C react-элементом в head", () => {
    const tree = renderer.create(<Circle head={"Head"} letter={""}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("С tail", () => {
    const tree = renderer.create(<Circle tail={"Tail"}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("C react-элементом в tail", () => {
    const tree = renderer.create(<Circle tail={"Tail"} letter={""}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("С index", () => {
    const tree = renderer.create(<Circle index={111}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("C пропом isSmall === true", () => {
    const tree = renderer.create(<Circle isSmall={true}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("В состоянии default", () => {
    const tree = renderer.create(<Circle state={ElementStates.Default}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("В состоянии changin", () => {
    const tree = renderer.create(<Circle state={ElementStates.Changing}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("В состоянии modified", () => {
    const tree = renderer.create(<Circle state={ElementStates.Modified}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});