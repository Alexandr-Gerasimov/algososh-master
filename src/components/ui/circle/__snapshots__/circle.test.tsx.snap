// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttons C react-элементом в head 1`] = `
<div
  className="content "
  data-cy="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
    data-cy="head"
  >
    Head
  </div>
  <div
    className="circle   default"
    data-cy="style"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="letter"
    >
      
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-cy="tail"
  />
</div>
`;

exports[`Buttons C react-элементом в tail 1`] = `
<div
  className="content "
  data-cy="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-cy="head"
  />
  <div
    className="circle   default"
    data-cy="style"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="letter"
    >
      
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
    data-cy="tail"
  >
    Tail
  </div>
</div>
`;

exports[`Buttons C пропом isSmall === true 1`] = `
<div
  className="content "
  data-cy="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-cy="head"
  />
  <div
    className="circle  small default"
    data-cy="style"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-cy="tail"
  />
</div>
`;

exports[`Buttons Без букв 1`] = `
<div
  className="content "
  data-cy="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-cy="head"
  />
  <div
    className="circle   default"
    data-cy="style"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-cy="tail"
  />
</div>
`;

exports[`Buttons В состоянии changin 1`] = `
<div
  className="content "
  data-cy="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-cy="head"
  />
  <div
    className="circle   changing"
    data-cy="style"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-cy="tail"
  />
</div>
`;

exports[`Buttons В состоянии default 1`] = `
<div
  className="content "
  data-cy="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-cy="head"
  />
  <div
    className="circle   default"
    data-cy="style"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-cy="tail"
  />
</div>
`;

exports[`Buttons В состоянии modified 1`] = `
<div
  className="content "
  data-cy="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-cy="head"
  />
  <div
    className="circle   modified"
    data-cy="style"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-cy="tail"
  />
</div>
`;

exports[`Buttons С head 1`] = `
<div
  className="content "
  data-cy="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
    data-cy="head"
  >
    Head
  </div>
  <div
    className="circle   default"
    data-cy="style"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-cy="tail"
  />
</div>
`;

exports[`Buttons С index 1`] = `
<div
  className="content "
  data-cy="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-cy="head"
  />
  <div
    className="circle   default"
    data-cy="style"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    111
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
    data-cy="tail"
  />
</div>
`;

exports[`Buttons С tail 1`] = `
<div
  className="content "
  data-cy="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-cy="head"
  />
  <div
    className="circle   default"
    data-cy="style"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
    data-cy="tail"
  >
    Tail
  </div>
</div>
`;

exports[`Buttons С буквами 1`] = `
<div
  className="content "
  data-cy="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-cy="head"
  />
  <div
    className="circle   default"
    data-cy="style"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="letter"
    >
      
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-cy="tail"
  />
</div>
`;
